src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitQueue.java

```diff
@@ -1,61 +1,69 @@
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
  */
 
 package com.exactpro.th2.common.schema.message.impl.rabbitmq;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Objects;
-import java.util.concurrent.atomic.AtomicReference;
-
-import org.jetbrains.annotations.NotNull;
-
+import com.exactpro.th2.common.schema.message.FilterFunction;
 import com.exactpro.th2.common.schema.message.MessageQueue;
 import com.exactpro.th2.common.schema.message.MessageSender;
 import com.exactpro.th2.common.schema.message.MessageSubscriber;
 import com.exactpro.th2.common.schema.message.configuration.QueueConfiguration;
 import com.exactpro.th2.common.schema.message.impl.rabbitmq.connection.ConnectionManager;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
 
 public abstract class AbstractRabbitQueue<T> implements MessageQueue<T> {
 
     private final AtomicReference<ConnectionManager> connectionManager = new AtomicReference<>();
     private final AtomicReference<QueueConfiguration> queueConfiguration = new AtomicReference<>();
+    private final AtomicReference<FilterFunction> filterFunc = new AtomicReference<>();
 
     private final AtomicReference<MessageSender<T>> sender = new AtomicReference<>();
     private final AtomicReference<MessageSubscriber<T>> subscriber = new AtomicReference<>();
 
     @Override
     public void init(@NotNull ConnectionManager connectionManager, @NotNull QueueConfiguration queueConfiguration) {
+        this.init(connectionManager, queueConfiguration, FilterFunction.DEFAULT_FILTER_FUNCTION);
+    }
+
+    @Override
+    public void init(@NotNull ConnectionManager connectionManager, @NotNull QueueConfiguration queueConfiguration, @NotNull FilterFunction filterFunc) {
         Objects.requireNonNull(connectionManager, "Connection can not be null");
         Objects.requireNonNull(queueConfiguration, "Queue configuration can not be null");
+        Objects.requireNonNull(filterFunc, "Filter function can not be null");
 
-        if (this.connectionManager.get() != null && this.queueConfiguration.get() != null) {
+        if (this.connectionManager.get() != null && this.queueConfiguration.get() != null && this.filterFunc.get() != null) {
             throw new IllegalStateException("Queue is already initialize");
         }
 
         this.connectionManager.set(connectionManager);
         this.queueConfiguration.set(queueConfiguration);
+        this.filterFunc.set(filterFunc);
```

Comment:
Maybe reorder calls: if, set + null verification

Reply:
I think null verification should be before the set because if we get NPE on `queueConfiguration` or `filterFunc` we set the wrong data to previous fields.

Reply:
It is not actually because we migrated from init methods to a constructor in another branch.